external-providers/java-external-provider/pkg/java_external_provider/provider.go

```diff
@@ -884,3 +897,30 @@ func (p *javaProvider) GetDependencies(ctx context.Context) (map[uri.URI][]*prov
 func (p *javaProvider) GetDependenciesDAG(ctx context.Context) (map[uri.URI][]provider.DepDAGItem, error) {
 	return provider.FullDepDAGResponse(ctx, p.clients)
 }
+
+func (p *javaProvider) BuildSettingsFile(m2CacheDir string) (settingsFile string, err error) {
+	fileContentTemplate := `
+<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 https://maven.apache.org/xsd/settings-1.0.0.xsd">
+  <localRepository>%v</localRepository>
+</settings>
+	`
+	var settingsFilePath string
+	m2Home := os.Getenv("M2_HOME")
```

Comment:
I think I don't understand why M2_HOME is used

Reply:
Got it from https://maven.apache.org/settings.html